src/components/TheControlFields.vue

```diff
@@ -117,61 +102,36 @@ export default {
         true,
         true,
       ],
-      img_url: "http://thacer.archaiodata.com/ThaCER.svg", // test images
     };
   },
   computed: {
-    // liste les groupes pour l'accordéon
+    // liste les groupes pour l'accordéon des champs
     // info : la liste des fields par groupe se trouvent dans types.groups.fields
+    // on pourrait s'assurer ici que le label de la langue existe toujours pour eviter la methode "labels"
     groups() {
-      if (this.allTypes) {
-        return this.allTypes.filter((x) => {
-          return x.type.includes(this.selectedtype);
-        })[0].groups;
-      } else {
-        return this.types.filter((x) => {
-          return x.type.includes(this.selectedtype);
-        })[0].groups;
-      }
-    },
-
-    // liste tous les fields afin ensuite d'établi la liste des labels correspondants
-    allfields() {
-      // TODO: warning ! There is global allFields, and this local allfields
-      if (this.allFields) {
-        return this.allFields.map(({ field }) => {
-          return field;
-        });
-      } else {
-        return this.fields.map(({ field }) => {
-          return field;
-        });
-      }
+      return this.allTypes.filter((x) => {
+        return x.type.includes(this.selectedType);
+      })[0].groups;
     },
 
-    alllabels() {
-      if (this.allFields) {
-        return this.allFields.map((field) => {
-          if (Object.prototype.hasOwnProperty.call(field, "labels")) {
-            return field.labels[this.appState.lang];
-          } else {
-            return field.field;
-          }
-        });
-      } else {
-        return this.fields.map((field) =>
-          field.labels !== undefined ? field.labels.en : field.field
-        );
-      }
+    allFieldsLabel() {
+      return this.allFields.map((field) => {
+        if (Object.prototype.hasOwnProperty.call(field, "labels")) {
```

Comment:
@AlZorba pour une prochaine fois : maintenant on peut faire `if ("labels" in field) {`
Plus elegant